go.mod

```diff
@@ -1011,3 +1044,14 @@ exclude github.com/docker/distribution v2.8.0+incompatible
 // some dependencies attempt to bring something like v1.8.2-0.20220303173753-edfe657b5405, which is older than v0.38.0
 // at the time of this inclusion, v0.38.0 was the latest version available (also tagged as v2.38.0)
 replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.38.0
+
+replace github.com/hpcloud/tail => github.com/grafana/tail v0.0.0-20220426200921-98e8eb28ea4c
+
+// using fork for support LogpullReceived and LogpullFields for loki - see https://github.com/cloudflare/cloudflare-go/pull/743
+replace github.com/cloudflare/cloudflare-go => github.com/cyriltovena/cloudflare-go v0.27.1-0.20211118103540-ff77400bcb93
+
+// using fork for custom dialer support - see https://github.com/bradfitz/gomemcache/pull/86
+replace github.com/bradfitz/gomemcache => github.com/grafana/gomemcache v0.0.0-20220812141859-1e3ae89e91a7
+
+// Fork containing a line-buffered logger which should improve logging performance.
+replace github.com/go-kit/log => github.com/dannykopping/go-kit-log v0.2.2-0.20221002180827-5591c1641b6b
```

Comment:
It seems odd that one component would require us to almost double the number of special cases in this file. Is there a general reason that this would necessarily be the case? Is it possible to remove any of these?

Reply:
That's a good observation. I removed the last replace (github.com/go-kit/log) by downgrading loki a little bit. I can't remove other `replace`s right now because loki needs them. They will be eventually removed when they will be removed from loki